stage0/src/acpi.rs

```diff
@@ -114,6 +116,118 @@ impl Rsdp {
 
         Ok(())
     }
+
+    pub fn xsdt(&self) -> Result<Option<&Xsdt>, &'static str> {
+        if self.xsdt_address == 0 {
+            Ok(None)
+        } else {
+            Some(Xsdt::new(VirtAddr::new(self.xsdt_address))).transpose()
```

Comment:
TIL about transpose :). Nit, not sure if it's the super readable here, since the option is always `Some`. I probs would've used the operator expression on `Xsdt::new` instead but don't feel strongly about it

Reply:
The situation I want to capture here is:

`Ok(None)` -- there was no XSDT
`Ok(Some(xsdt))` -- it was parsed successfully
`Err(err)` -- there was some error finding or validating the XSDT table

I went back and forth which way it makes more sense to have the `Some` and `Result`, and I think this way makes the most sense.

The other way around it'd be:
`None` - no XSDT
`Some(Ok(xsdt))` - parsing successful
`Some(Err(err))` -- error

...and I think it'd be worse.

Reply:
I think the type signature of the return value is good. 

I'd just find to be more idiomatic to express it as:

```
if self.xsdt_address == 0 {
            Ok(None)
        } else {
            let xsdt = Xsdt::new(VirtAddr::new(self.xsdt_address))?;
            Some(xsdt)
}
```

Reply:
the transpose call in the else seems a bit overkill, since the Option it is called upon will never be None.

Reply:
I changed it to a `.map(Some)` here, this is probably indeed cleaner than `transpose()`.